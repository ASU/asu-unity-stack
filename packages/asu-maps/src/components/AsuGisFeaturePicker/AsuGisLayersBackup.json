[{"id":"CampusTransit_2204","layerType":"ArcGISMapServiceLayer","url":"https:\/\/gis.m.asu.edu\/server\/rest\/services\/Campus\/CampusTransit\/MapServer","visibility":false,"opacity":1,"title":"Transit","itemId":"824610651f01450483eb3c1e20e3c607","value":"CampusTransit_2204","label":"Transit","children":[{"id":0,"name":"ASU Cart Daytime Parking ","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_0","label":"ASU Cart Daytime Parking "},{"id":1,"name":"ASU Cart Paths","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_1","label":"ASU Cart Paths"},{"id":3,"name":"Bike Parking","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_3","label":"Bike Parking"},{"id":4,"name":"Bike Repair","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_4","label":"Bike Repair"},{"id":5,"name":"Car Share","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_5","label":"Car Share"},{"id":6,"name":"Flash Shuttle","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_6","label":"Flash Shuttle"},{"id":7,"name":"Intercampus Shuttle stops","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_7","label":"Intercampus Shuttle stops"},{"id":9,"name":"Light Rail Stops","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_9","label":"Light Rail Stops"},{"id":10,"name":"Park-and-Rides","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_10","label":"Park-and-Rides"},{"id":11,"name":"Parking","value":"CampusTransit_2204_11","label":"Parking"},{"id":12,"name":"Accessible Parking","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_12","label":"Accessible Parking"},{"id":13,"name":"EV Charging","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_13","label":"EV Charging"},{"id":15,"name":"Permit","layerDefinition":{"definitionExpression":"(Visible = 1) OR (Visible = 0) OR (Visible IS NULL)"},"value":"CampusTransit_2204_15","label":"Permit"},{"id":16,"name":"Visitor","value":"CampusTransit_2204_16","label":"Visitor"},{"id":17,"name":"DayPass","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_17","label":"DayPass"},{"id":18,"name":"Event","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_18","label":"Event"},{"id":19,"name":"Gated","layerDefinition":{"definitionExpression":"Visible = 1 OR Visible = 0"},"value":"CampusTransit_2204_19","label":"Gated"},{"id":20,"name":"Pay Station","layerDefinition":{"definitionExpression":"Visible = 1 OR Visible = 0"},"value":"CampusTransit_2204_20","label":"Pay Station"},{"id":21,"name":"Valet","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_21","label":"Valet"},{"id":24,"name":"Scooter Parking","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_24","label":"Scooter Parking"},{"id":25,"name":"Scooter Boundary","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_25","label":"Scooter Boundary"},{"id":26,"name":"Walk-only Zones","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusTransit_2204_26","label":"Walk-only Zones"}]},{"id":"CampusSustainability_2508","layerType":"ArcGISMapServiceLayer","url":"https:\/\/gis.m.asu.edu\/server\/rest\/services\/Campus\/CampusSustainability\/MapServer","visibility":false,"opacity":1,"title":"Sustainability","itemId":"212f9f46185f4012a00f958b427cfd42","value":"CampusSustainability_2508","label":"Sustainability","children":[{"id":0,"name":"EV Charging","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusSustainability_2508_0","label":"EV Charging"},{"id":1,"name":"Gardens","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusSustainability_2508_1","label":"Gardens"},{"id":2,"name":"Renewables","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusSustainability_2508_2","label":"Renewables"}]},{"id":"CampusServices_4160","layerType":"ArcGISMapServiceLayer","url":"https:\/\/gis.m.asu.edu\/server\/rest\/services\/Campus\/CampusServices\/MapServer","visibility":false,"opacity":1,"title":"Services","itemId":"6a569ff59b224291b4fd1fad3dd47120","value":"CampusServices_4160","label":"Services","children":[{"id":0,"name":"AED","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_0","label":"AED"},{"id":1,"name":"Computing Center","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_1","label":"Computing Center"},{"id":2,"name":"Dining","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_2","label":"Dining"},{"id":3,"name":"Emergency Call Box","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_3","label":"Emergency Call Box"},{"id":4,"name":"Gender-Neutral Restroom","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_4","label":"Gender-Neutral Restroom"},{"id":5,"name":"Health","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_5","label":"Health"},{"id":6,"name":"Information","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_6","label":"Information"},{"id":7,"name":"Library","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_7","label":"Library"},{"id":8,"name":"Merchandise","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_8","label":"Merchandise"},{"id":9,"name":"Museum","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_9","label":"Museum"},{"id":10,"name":"Police","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_10","label":"Police"},{"id":11,"name":"Tutoring Centers","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_11","label":"Tutoring Centers"},{"id":12,"name":"Universal Changing Table","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_12","label":"Universal Changing Table"},{"id":13,"name":"Wellness\/Lactation Rooms","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusServices_4160_13","label":"Wellness\/Lactation Rooms"}]},{"id":"CampusPointsofPride_4284","layerType":"ArcGISMapServiceLayer","url":"https:\/\/gis.m.asu.edu\/server\/rest\/services\/Campus\/CampusPointsofPride\/MapServer","visibility":false,"opacity":1,"title":"Points of Pride","itemId":"e24b22376f3344f09bb1c0cb4dd89ad4","value":"CampusPointsofPride_4284","label":"Points of Pride","children":[{"id":0,"name":"Fraternity \u0026 Sorority Life","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusPointsofPride_4284_0","label":"Fraternity \u0026 Sorority Life"},{"id":1,"name":"Public Art","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusPointsofPride_4284_1","label":"Public Art"},{"id":2,"name":"Tempe Arboretum","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusPointsofPride_4284_2","label":"Tempe Arboretum"},{"id":3,"name":"Other Points of Pride","layerDefinition":{"definitionExpression":"Visible = 1"},"value":"CampusPointsofPride_4284_3","label":"Other Points of Pride"},{"id":4,"name":"Malls","layerDefinition":{"definitionExpression":"Created_By IS NOT NULL"},"value":"CampusPointsofPride_4284_4","label":"Malls"}]},{"id":"CampusBuilding_7147","layerType":"ArcGISMapServiceLayer","url":"https:\/\/gis.m.asu.edu\/server\/rest\/services\/Campus\/CampusBuilding\/MapServer","visibility":false,"opacity":1,"title":"Buildings","itemId":"f1378070504a4191b3a8ef2cfdaea346","showLegend":true,"value":"CampusBuilding_7147","label":"Buildings","children":[{"id":1,"name":"Academic","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":1},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_1","label":"Academic"},{"id":2,"name":"Administrative","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":2},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_2","label":"Administrative"},{"id":3,"name":"Athletics","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":3},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_3","label":"Athletics"},{"id":4,"name":"Fine Arts","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":4},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_4","label":"Fine Arts"},{"id":5,"name":"Housing","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":5},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_5","label":"Housing"},{"id":6,"name":"Library","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":6},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_6","label":"Library"},{"id":7,"name":"Off-Campus","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":7},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_7","label":"Off-Campus"},{"id":8,"name":"Research","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":8},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_8","label":"Research"},{"id":9,"name":"Student Services","layerDefinition":{"source":{"type":"mapLayer","mapLayerId":9},"definitionExpression":"Visible = 1"},"minScale":0,"maxScale":0,"parentLayerId":-1,"defaultVisibility":false,"value":"CampusBuilding_7147_9","label":"Student Services"}]}]
