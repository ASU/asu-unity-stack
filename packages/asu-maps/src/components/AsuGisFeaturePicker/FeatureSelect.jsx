import React, { useState } from "react";
import PropTypes from "prop-types";

const FeatureSelect = props => {
  // we need to store parent layer id along
  // with the feature object id in order to build
  // interactive map URL
  const parent = props.parent;
  const options = props.options;

  // initialize select state from passed selected prop
  const [selected, setSelect] = useState(props.selected);

  // handles changes in select list
  const handleChange = e => {
    const val = e.target.value;

    setSelect(val);

    if (props.onChange) {
      props.onChange(val);
    }
  };

  if (options.length > 0) {
    return (
      <div style={{ paddingLeft: "75px" }}>
        <label>
          <h2>Select a Feature</h2>
          <select value={selected} onChange={e => handleChange(e)}>
            <option value="{}">Choose...</option>
            {options
              .sort((a, b) => {
                if (a.attributes.Name < b.attributes.Name) {
                  return -1;
                }
                if (a.attributes.Name > b.attributes.Name) {
                  return 1;
                }
                return 0;
              })
              .map((item, index) => {
                if (item.attributes.OBJECTID && item.attributes.Name) {
                  // reshape the feature data from the API
                  const data = {
                    id: item.attributes.OBJECTID,
                    name: item.attributes.Name,
                    lat: item.geometry.y,
                    lng: item.geometry.x,
                    parent: parent
                  };

                  return (
                    <option
                      key={item.attributes.OBJECTID}
                      value={JSON.stringify(data)}
                    >
                      {item.attributes.Name}
                    </option>
                  );
                }
              })}
          </select>
        </label>
      </div>
    );
  } else {

    const noResStyles = {
      padding: '2em'
    };

    return (
      <div style={noResStyles}>
        <h2>No options to display.</h2>
      </div>
    );
  }
};

FeatureSelect.propTypes = {
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func,
  parent: PropTypes.string,
  selected: PropTypes.string
};

FeatureSelect.defaultProps = {
  options: [],
  selected: "{}"
};

export default FeatureSelect;
