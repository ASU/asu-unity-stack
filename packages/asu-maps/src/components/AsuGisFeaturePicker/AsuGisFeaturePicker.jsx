import React, { useState, useEffect, useCallback, useRef } from "react";
import PropTypes from "prop-types";
import defaultLayers from "./AsuGisLayersBackup.json";
import CheckboxTree from "react-checkbox-tree";
import FeatureSelect from "./FeatureSelect";
import "react-checkbox-tree/lib/react-checkbox-tree.css";

// Get the  numerical layer ID given a tree node from the
// react-checkbox-tree component
const getParentId = (layers, value) => {
  let id;

  if (value) {
    for (let i = 0; i < layers.length; i++) {
      if (value.includes(layers[i].id)) {
        const topLayer = layers[i];
        id = topLayer.id;
      }
    }
  }

  return id;
};

/** Get the layer API URL given the tree and a selected tree value */
const getLayerUrl = (layers, value) => {
  let url;
  const query = "/query?where=1=1&outFields=OBJECTID,Name,SHAPE&f=json";

  for (let i = 0; i < layers.length; i++) {
    if (value.includes(layers[i].id) && value != layers[i].id) {
      const topLayer = layers[i];
      const childId = value.substring(topLayer.id.length + 1);
      url = topLayer.url + "/" + childId + query;
    }
  }

  return url;
};

/**
 * Main parent component which renders entire ASU Maps GIS Feature Select UI
 * @param {*} props
 */
const AsuGisFeaturePicker = props => {
  // Layer data and state hooks for the
  // GIS map layer tree select
  const layers = props.layers;

  const savedLayer = props.selected.parent ? [props.selected.parent] : [];
  const parentLayer =
    savedLayer.length > 0 ? [getParentId(layers, savedLayer[0])] : [];

  // These are required in order to control the react-checkbox-tree
  // component
  const [checked, setChecked] = useState(savedLayer);
  const [expanded, setExpanded] = useState(parentLayer);

  // State hooks for the GIS feature select list component
  const [selectState, setSelectState] = useState({
    status: "noselection",
    options: [],
  });

  // Callbacks for setting state of the react-checkbox-tree tree component and
  // saving the tree node for later use.
  const checkCallback = useCallback(
    (checked, targetNode) => {
      // Set checked state to only the latest checked node value. This logic
      // in effect only allows one selected value at a time.
      if (checked.length > 0) {
        setChecked([targetNode.value]);

        // if checked is empty (unchecked), set to the empty val
      } else {
        setChecked(checked);
      }

      // run the onChange function with an empty select option value since
      // swapping out the Select list got rid of the old value
      const emptyVal = "{}";
      if (props.onChange) {
        props.onChange(emptyVal);
      }
    },
    [checked]
  );

  const expandCallback = useCallback(
    (expanded, targetNode) => {
      setExpanded(expanded);
    },
    [expanded]
  );

  //This effect only runs when the selected map layer changes.
  useEffect(() => {
    /* Helper function to do fetch request to the GIS layer
    endpoint, then sets the 'options' state variable with
    retrieved items. This in turn causes the select list to
    re-render with new options.*/
    const getOptions = async value => {
      let result = [];
      let error = false;

      if (value) {
        // Build query URL given the currently checked value and layer tree
        const url = getLayerUrl(props.layers, value);

        // Fetch map features from GIS api and set options state using setOptions hook
        if (url) {
          result = await fetch(url)
            .then(res => res.json())
            .then(
              result => {
                if (result.features) {
                  // set the options state.
                  return result.features;
                } else {
                  // set empty if no features returned from result
                  return [];
                }
              },
              // Note: it's important to handle errors here
              // instead of a catch() block so that we don't swallow
              // exceptions from actual bugs in components.
              error => {
                error = true;
                console.error(error);
                return [];
              }
            );
        }

        if (error) {
          setSelectState({
            status: "error",
            options: [],
          });
        } else {
          setSelectState({
            status: "success",
            options: result,
          });
        }
      }
    };

    // set state to 'loading' before getting options
    setSelectState({
      status: "loading",
      options: [],
    });

    getOptions(checked[0]);
  }, [checked]);

  // Render the select box or messages depending
  // on current status
  const renderSelect = (selectState, props) => {
    if (selectState.status == "error") {
      return <div>Something went wrong</div>;
    }

    if (selectState.status == "loading") {
      return <div>Loading...</div>;
    }

    return (
      <FeatureSelect
        {...{
          parent: checked.length > 0 ? checked[0] : undefined,
          options: selectState.options,
          status: selectState.status,
          onChange: props.onChange,
          selected: props.selected ? JSON.stringify(props.selected) : undefined,
        }}
      />
    );
  };

  // Apply container style inline for now.
  // As we theme this component in the future we may want to add a modular CSS
  // stylesheet as well.
  const containerStyle = {
    display: "flex",
    flexDirection: "row",
  };

  // return complete feature selection widget
  return (
    <div style={containerStyle}>
      <div>
        <CheckboxTree
          checked={checked}
          expanded={expanded}
          onCheck={checkCallback}
          onExpand={expandCallback}
          nodes={layers}
          nativeCheckboxes={true}
          onlyLeafCheckboxes={true}
          id="asu-gis-fp"
        />
      </div>
      <div>
        {checked.length > 0 ? (
          renderSelect(selectState, props)
        ) : (
          <h2 style={{ paddingLeft: "1em" }}>
            &#8592; Select a layer to display options.
          </h2>
        )}
      </div>
    </div>
  );
};

AsuGisFeaturePicker.propTypes = {
  layers: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func,
  selected: PropTypes.object,
};

AsuGisFeaturePicker.defaultProps = {
  layers: defaultLayers,
  selected: {},
};

export default AsuGisFeaturePicker;
