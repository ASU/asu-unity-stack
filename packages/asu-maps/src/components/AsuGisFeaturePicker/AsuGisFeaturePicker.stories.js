import React from "react";

import AsuGisFeaturePicker from "./index.js";
import { withKnobs, object } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";

export default {
  component: AsuGisFeaturePicker,
  title: "AsuGisFeaturePicker",
  decorators: [withA11y, withKnobs]
};

export const basic = () => <AsuGisFeaturePicker />;

export const withSelection = () => (
  <AsuGisFeaturePicker
    selected={object("selected", {
      id: 618,
      name: "Mercado F",
      lat: -112.06557500040057,
      lng: 33.45072599973827,
      parent: "CampusBuilding_7147_1"
    })}
  />
);
