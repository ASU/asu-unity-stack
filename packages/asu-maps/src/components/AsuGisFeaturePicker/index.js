import AsuGisFeaturePicker from "./AsuGisFeaturePicker";

export default AsuGisFeaturePicker;
