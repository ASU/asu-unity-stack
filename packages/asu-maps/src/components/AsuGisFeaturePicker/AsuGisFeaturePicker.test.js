import React from 'react'
import { render} from '@testing-library/react'
import AsuGisFeaturePicker from './AsuGisFeaturePicker'



describe('The AsuGisFeaturePicker', () => {
    it('renders the component', () => {

      const {container} = render(<AsuGisFeaturePicker/>);
            expect(container.firstChild).toMatchSnapshot();
    })
});
