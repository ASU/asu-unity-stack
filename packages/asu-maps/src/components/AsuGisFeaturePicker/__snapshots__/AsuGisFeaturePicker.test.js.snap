// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AsuGisFeaturePicker renders the component 1`] = `
<div
  style="display: flex; flex-direction: row;"
>
  <div>
    <div
      class="react-checkbox-tree rct-icons-fa4 rct-native-display"
      id="asu-gis-fp"
    >
      <ol>
        <li
          class="rct-node rct-node-parent rct-node-collapsed"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            >
              <span
                class="rct-icon rct-icon-expand-close"
              />
            </button>
            <span
              class="rct-bare-label"
              title="Transit"
            >
              <span
                class="rct-node-icon"
              >
                <span
                  class="rct-icon rct-icon-parent-close"
                />
              </span>
              <span
                class="rct-title"
              >
                Transit
              </span>
            </span>
          </span>
        </li>
        <li
          class="rct-node rct-node-parent rct-node-collapsed"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            >
              <span
                class="rct-icon rct-icon-expand-close"
              />
            </button>
            <span
              class="rct-bare-label"
              title="Sustainability"
            >
              <span
                class="rct-node-icon"
              >
                <span
                  class="rct-icon rct-icon-parent-close"
                />
              </span>
              <span
                class="rct-title"
              >
                Sustainability
              </span>
            </span>
          </span>
        </li>
        <li
          class="rct-node rct-node-parent rct-node-collapsed"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            >
              <span
                class="rct-icon rct-icon-expand-close"
              />
            </button>
            <span
              class="rct-bare-label"
              title="Services"
            >
              <span
                class="rct-node-icon"
              >
                <span
                  class="rct-icon rct-icon-parent-close"
                />
              </span>
              <span
                class="rct-title"
              >
                Services
              </span>
            </span>
          </span>
        </li>
        <li
          class="rct-node rct-node-parent rct-node-collapsed"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            >
              <span
                class="rct-icon rct-icon-expand-close"
              />
            </button>
            <span
              class="rct-bare-label"
              title="Points of Pride"
            >
              <span
                class="rct-node-icon"
              >
                <span
                  class="rct-icon rct-icon-parent-close"
                />
              </span>
              <span
                class="rct-title"
              >
                Points of Pride
              </span>
            </span>
          </span>
        </li>
        <li
          class="rct-node rct-node-parent rct-node-collapsed"
        >
          <span
            class="rct-text"
          >
            <button
              aria-label="Toggle"
              class="rct-collapse rct-collapse-btn"
              title="Toggle"
              type="button"
            >
              <span
                class="rct-icon rct-icon-expand-close"
              />
            </button>
            <span
              class="rct-bare-label"
              title="Buildings"
            >
              <span
                class="rct-node-icon"
              >
                <span
                  class="rct-icon rct-icon-parent-close"
                />
              </span>
              <span
                class="rct-title"
              >
                Buildings
              </span>
            </span>
          </span>
        </li>
      </ol>
    </div>
  </div>
  <div>
    <h2
      style="padding-left: 1em;"
    >
      ‚Üê Select a layer to display options.
    </h2>
  </div>
</div>
`;
